core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -552,7 +552,9 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
       taskId: String,
       reason: String): Unit = {
     stateLock.synchronized {
-      removeExecutor(taskId, SlaveLost(reason))
+      if (!stopCalled) {
```

Comment:
I don't think we should be adding the guard here.  It's the parent class that's incorrectly making a request to the `driverEndpoint` despite the `driverEndpoint` being shut down.  So it's the parent class that should add the guard.